js/client-mainmenu.js

```diff
@@ -114,7 +125,68 @@
 				);
 			}
 		},
+		createLoginFormObserver:function (self) {
+			const config = {attributes: true, childList: true, subtree: true};
+			const observer = new MutationObserver(function () {
+				if (window.app.user.get("named") && window.Storage.prefs("page") == "challenge") {
+					window.Storage.prefs("page", "testclient");
+					if ($('.challenge-container').length == 0) {
+						//user log in event trigerd when user log in and make user pick a team
+						$('body').on("login", function () {
+							self.onUserLogIn(self);
+						});
+						$('body').trigger("login");
+					}
+				}
+			});
+			observer.observe(document.activeElement, config);
+		},
+		onUserLogIn:function (self) {
+			let userId = window.Storage.prefs("challenger");
+			let format = window.Storage.prefs("format");
+			app.rooms[''].challenge(userId, format, (format.includes("random") ? '' : format));
+			$('.select.formatselect').prop("disabled",true);
+		},
+		generateLink:function (userId, format) {
+			if (!$('.link').length) {
+				let link = window.location.protocol+"//" + window.Config.routes.client + "?challenge-" + userId + "-" + format;
+				$('.link-generatore').append("<p class=link>" + link + "</p>");
+				$('#generate-link').html("generate link");
```

Comment:
```suggestion
				$('#generate-link').html("generate link");
```
This should be using element names, not ids.